### PR TITLE
Added labels, Deployment and tidied up HELM templates

### PR DESCRIPTION
**Pull request by noelo**

### CODE REVIEW

File: chart/templates/_helpers.tpl [hunk 1]
```diff
@@ -58,16 +63,19 @@ helm.sh/chart: {{ include "pet-battle-tournament.chart" . }}
...
 {{/*
 Selector labels
 */}}
+
+
 {{- define "pet-battle-tournament.selectorLabels" -}}
 app.kubernetes.io/name: {{ include "pet-battle-tournament.name" . }}
 app.kubernetes.io/component: {{ include "pet-battle-tournament.name" . }}
 app.kubernetes.io/instance: {{ .Release.Name }}
-deploymentconfig: {{ include "pet-battle-tournament.fullname" . }}
```
Comment (springdo) [inline chart/templates/_helpers.tpl:70]: There is some control flow in your `values.yaml` for DC on or off. Will this break the things if I now select DC on?

File: chart/templates/_helpers.tpl [hunk 2]
```diff
@@ -58,16 +63,19 @@ helm.sh/chart: {{ include "pet-battle-tournament.chart" . }}
 app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
 {{- end }}
 app.kubernetes.io/managed-by: {{ .Release.Service }}
+app.kubernetes.io/part-of: petbattleworld
+{{ include "pet-battle-tournament.stateLabels" . }}
```
Comment (springdo) [inline chart/templates/_helpers.tpl]: is this stateLabels thing meant to be empty?

File: chart/templates/_helpers.tpl (hunk 1 above)
Comment (noelo) [reply chart/templates/_helpers.tpl:70]: yea I couldn't figure out how to do not set in helm e.g. if DC is true then ignore the deployement and vice veras

**Comment by eformat** [comment]
try example here:

https://github.com/ActiveLearningStudio/rh-innovation-lab-open-shift/blob/main/curriki-api/chart/templates/build-config.yaml#L1

**Pull request merged**